marketplacetests/firefox_accounts/app.py

```diff
@@ -50,5 +71,20 @@ def tap_next(self):
 
     def tap_sign_in(self):
         self.keyboard.dismiss()
+        self.change_to_correct_login_frame()
```

Comment:
I wouldn't save the frame after switching to it, rather I would keep the same logic that is used now to figure out which frame to switch to (which is done before switching), and save that information.